src/formats/ff.c

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+// farbfeld format decoder
+
+#include "../loader.h"
+
+#include <limits.h>
+#include <stdint.h>
+
+
+enum loader_status decode_ff(struct image* ctx, const uint8_t* data,
+                              size_t size)
+{
+    size_t width, height;
+
+    if (size < 16 ) {
+        return ldr_unsupported;
+    }
+
+    uint8_t mByte[8] = {'f', 'a', 'r', 'b', 'f', 'e', 'l', 'd'};
+
+    for (int i = 0; i < 8; i++) if (data[i] != mByte[i]) return ldr_unsupported;
+
+    width = data[11] + data[10]*256 + data[9]*65536 + data[8]*1638400;
+    height = data[15] + data[14]*256 + data[13]*65536 + data[12]*1638400;
+
+    ctx->alpha=true;
+
+    if (!image_allocate_frame(ctx, width, height)) return ldr_fmterror;
+
+    struct pixmap* pm = &ctx->frames[0].pm;
+
+    for (size_t i = 16; i < size; i=i+8) {
+        argb_t pixel = ARGB(data[i+6], data[i], data[i+2], data[i+4]);
+	pm->data[(i-16)/8] = pixel;
```

Comment:
Segfault if file is corrupted, e.g. `width * height * sizeof(argb) < size`.

Reply:
Oh.... Let's simulate =)
```
with = 10;
height = 10;
size = 10000000000;
```

Reply:
I have tried both exceeding size (`size > 16+width*height*sizeof(argb_t)*2`) and insufficient size (`size < 16+width*height*sizeof(argb_t)*2`)

In the case of insufficient size, as expected, fails the line 27 test. In case of exceeding size, as expected, displays garbage. No segfault in either case.

The exceeding size is matter of policy: it's better to display a broken image or to just fail the decode?

Reply:
It's better to display broken image.